src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -89,28 +90,51 @@ private static bool InternalExecute(AnalysisConfig config, IEnumerable<string> u
                 return false;
             }
 
-            var exeFileName = FindScannerExe();
+            var exeFileName = FindScannerExe(logger, Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location));
             return ExecuteJavaRunner(config, userCmdLineArguments, logger, exeFileName, fullPropertiesFilePath, new ProcessRunner(logger));
         }
 
-        private static string FindScannerExe()
+        internal /* for testing */ static string FindScannerExe(ILogger logger, string scannerCliDirectoryLocation, string scannerVersion = SonarScannerVersion)
         {
-            var binFolder = Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location);
-            var fileExtension = PlatformHelper.IsWindows() ? ".bat" : "";
-            return Path.Combine(binFolder, $"sonar-scanner-{SonarScannerVersion}", "bin", $"sonar-scanner{fileExtension}");
+            var scannerCliFolder = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-{scannerVersion}");
+
+            // During packaging of the artifacts (see script https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/scripts/package-artifacts.ps1)
+            // the scanner-cli is unzipped for .NET Framework 4.6, but not for the .NET and .NET Core - where it's unzipped upon first usage of the scanner.
+            // For this reason, the 'if' block below will not be executed for .NET Framework.
+            if (!Directory.Exists(scannerCliFolder))
+            {
+                // We unzip the scanner-cli-{version}.zip while in the user's machine so that the Unix file permissions are not lost.
+                // The unzipping happens only once, during the first scanner usage.
+                var zipPath = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-cli-{scannerVersion}.zip");
```

Comment:
maybe extract `sonar-scanner-cli-{scannerVersion}.zip` to a string, it is used in multiple locations